architecture/modules/functions/variables.tf

```diff
@@ -9,4 +9,5 @@ variable "elpis_worker" {}
 variable "user_upload_files_bucket" {}
 variable "user_datasets_bucket" {}
 
-variable "dataset_processing_topic" {}
\ No newline at end of file
+variable "dataset_processing_topic" {}
```

Comment:
The comments you left on the terraform side of things made me realise it's probably worth taking a harder look at the structure of the architecture folder, adding types and documentation and formatting, so I've created another branch to do so here: https://github.com/CoEDL/elpiscloud/tree/architecture_improvements, in which I'll address these comments.

Reply:
Sadly, the variables weren't typed because I wasn't too familiar with terraform and that was what they had in the tutorials I was working through. With something like a third party pubsub topic, can I still use that as a type inside `variables.tf`?  Or should I be leaving  the more complex types as any?

Reply:
No worries. I'd honestly just suggest inlining all variables in the module files, there's really no reason for separate files. Define the types as far as practical and consider using outputs for the key values (ids, names, urls etc.) that you need to minimise the need for complex types, but if you still do need them, something like:
```
variable "stages" {
  type = map(object({
    storage_integration = string,
    url = string,
  }))
}
```
Is perfectly valid.